zerver/actions/message_edit.py

```diff
@@ -332,7 +332,10 @@ def get_mentions_for_message_updates(message_id: int) -> set[int]:
         )
         .values_list("user_profile_id", flat=True)
     )
-    return set(mentioned_user_ids)
+
+    user_ids_having_message_access = event_recipient_ids_for_action_on_messages([message])
+
+    return set(mentioned_user_ids) & user_ids_having_message_access
```

Comment:
> perform a bitwise AND with the mentioned users 

This is not a bitwise and -- just a normal and is what you want and have correctly implemented. "Bitwise and" means "3 & 7 = 3".

Reply:
Ah, I hope my CS-101 prof don't see this 😄 

My mind just calls `&` , `|` as bitwise operators (C++ thing...)

Fixing commit message.